src/PathUtilities.cpp

```diff
@@ -0,0 +1,302 @@
+            StrIt = std::next(StrIt,2);
+        }
+        while( StrIt != ToCheck.end() )
+        {
+            if( (*StrIt) == '/' || (*StrIt) == '\\' ) {
+                if( DirStr == ".." ) {
+                    Depth -= 1;
+                }else if( !DirStr.empty() && DirStr != "." ) {
+                    Depth += 1;
+                }
+                DirStr.clear();
+
+                if( ExitIfNegative && Depth < 0 ) {
+                    break;
+                }
+            }else{
+                DirStr.append(1,*StrIt);
+            }
+            ++StrIt;
+        }
+        return Depth;
+    }
+
+    Boole IsSubPath(const StringView BasePath, const StringView CheckPath)
+    {
+        Boole BaseIsAbsolute = IsPathAbsolute(BasePath);
+        Boole CheckIsAbsolute = IsPathAbsolute(CheckPath);
+
+        if( BaseIsAbsolute && !CheckIsAbsolute ) {
+            throw std::runtime_error("Attempting to compare absolute base path with relative subpath.");
+        }
+        if( !BaseIsAbsolute && CheckIsAbsolute ) {
+            throw std::runtime_error("Attempting to compare relative base path with absolute subpath.");
+        }
+
+        String NormBasePath = RemoveDotSegments(BasePath);
+        String NormCheckPath = RemoveDotSegments(CheckPath);
+        String::const_iterator BaseIt = NormBasePath.begin();
+        String::const_iterator BaseEnd = NormBasePath.end();
+        String::const_iterator CheckIt = NormCheckPath.begin();
+        String::const_iterator CheckEnd = NormCheckPath.end();
+
+        if( IsPathAbsolute_Windows(NormBasePath) ) {
+            std::advance(BaseIt,2);
+        }
+        if( IsPathAbsolute_Windows(NormCheckPath) ) {
+            std::advance(CheckIt,2);
+        }
+
+        while( BaseIt != BaseEnd )
+        {
+            if( CheckIt == CheckEnd || (*BaseIt) != (*CheckIt) ) {
+                return false;
+            }
+
+            ++BaseIt;
+            ++CheckIt;
+        }
+
+        String CheckRemains(CheckIt,CheckEnd);
+        return ( GetDirectoryDepth(CheckRemains,true) > 0 );
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Path Utilities
+
+    String BuildPath(const StringView PathRoot, const StringVector& ToBuild,
+                     const StringView FileName, const Boole UseWindowsSlash)
+    {
+        // Char count gets used to reserve space on the return string
+        size_t CharCount = PathRoot.size() + ToBuild.size() + FileName.size();
+        StringVector::const_iterator BuildIt = ToBuild.begin();
+        std::for_each(BuildIt,ToBuild.cend(),[&CharCount](const String& CurrStr) {
+            CharCount += CurrStr.length();
+        });
+        String Ret;
+        Ret.reserve(CharCount);
+        Ret.append(PathRoot);
+        const Char8 Slash = UseWindowsSlash ? GetDirectorySeparator_Windows() : GetDirectorySeparator_Posix();
+        // Build the path
+        while( BuildIt != ToBuild.end() )
```

Comment:
StringTools::join?

Reply:
Didn't exist when I originally wrote this.  I'll evaluate it to see if I can drop it in.